### PR TITLE
fix(Core/Item): Fixed possible dupe with container items containing non un…

### PR DESCRIPTION
**Pull request by robinsch**
…ique non stackable items

## Changes Proposed:
-  Delete the correct item from item_loot_items - instead of a random one. If item container multiple items it would result in a dupe, refilling the item on startup.
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/6939

## Tests Performed:

- This has been tested on a TC-based source code with many players online

## How to Test the Changes:

- Check the way items are stored, for example try to create a character, get some items, try to restart the server and make sure there is no regression and everything is normal with the stored items

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_characters/rev_1626271703991541000.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `version_db_characters` (`sql_rev`) VALUES ('1626271703991541000');
+ALTER TABLE `item_loot_storage` 
+ADD COLUMN `item_index` int(10) UNSIGNED NOT NULL DEFAULT 0 AFTER `count`;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_characters/rev_1626271703991541000.sql:3]:
```suggestion
  ADD COLUMN `item_index` INT UNSIGNED NOT NULL DEFAULT 0 AFTER `count`;
```

**Comment by pklloveyou** [comment]
I don't know what the vulnerability is, but it works normally after pr

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
good job @robinsch and thanks @pklloveyou  for testing this PR. Set it as [To Be Merged]

**Comment by Kitzunu** [comment]
But has it been tested with the vulnerability? pkl just said he didn't know what it was but that everything works as it should?

**Comment by FrancescoBorzi** [comment]
> But has it been tested with the vulnerability? pkl just said he didn't know what it was but that everything works as it should?

@robinsch will check tomorrow and will let us know

**Comment by pklloveyou** [comment]
It has been used on my server for several days, and no player has reported an error. How can the vulnerability recur

**Comment by FrancescoBorzi** [comment]
> It has been used on my server for several days, and no player has reported an error. How can the vulnerability recur

thank you very much @pklloveyou , I hope @robinsch can confirm soon that the vulnerability is gone with this

**Comment by robinsch** [comment]
Tested and works!

**Comment by LukasVolgger** [comment]
Thx @robinsch ! So i think this can be merged.

**Pull request merged**